presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -439,12 +439,18 @@ protected TupleDescriptor visitInsert(Insert insert, AnalysisContext context)
         analysis.setInsertTarget(targetTableHandle.get());
 
         List<ColumnMetadata> columns = metadata.getTableMetadata(session, targetTableHandle.get()).getColumns();
-        Iterable<Type> tableTypes = columns.stream()
+        Iterable<String> tableTypes = columns.stream()
```

Comment:
imo this is broken. See my patch which fixes the same thing.
You cannot just assume that if bases match we are good to go. Sometimes we need to add cast functions.